backend/routes/anomaly_routes.py

```
@@ -0,0 +1,102 @@
+            "by_type": by_type,
+            "by_location": by_location,
+            "generated_at": datetime.now(timezone.utc).isoformat(),
+        },
+    }
+
+
+@router.get("/all")
+async def get_all_anomalies(
+    limit: Optional[int] = Query(None, description="Max results; omit for all"),
+    offset: int = Query(0, description="Pagination offset"),
+    db: Session = Depends(get_db),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)